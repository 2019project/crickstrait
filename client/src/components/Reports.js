import React, { Component } from "react";
import "../styles/Reports.css";

export default class Reports extends Component {
  render() {
    return (
      <div className="financialTableContainer">
        <table className="financialTableData">
          {this.props.headers.map(indicatorName => (
            <tr>
              {indicatorName === "date" ? (
                <td className="financialTableDataTitle">Indicator Name</td>
              ) : indicatorName === "Total Assets" ||
                indicatorName === "Total Liabilities" ||
                indicatorName === "Equity Before Minorities" ||
                indicatorName === "Change in Working Capital" ? (
                <td className="financialsTotalTD">{indicatorName}</td>
              ) : (
                <td>{indicatorName}</td>
              )}
              <>
                {this.props.reportdata.map(value => (
                  <>
                    {indicatorName === "date" ? (
                      <td className="financialTableDataTitle">
                        {" "}
                        {new Date(value[indicatorName]).toLocaleDateString(
                          "en-IN",
                          {
                            month: "short",
                            year: "2-digit"
                          }
                        )}
                      </td>
                    ) : indicatorName === "Total Assets" ||
                      indicatorName === "Total Liabilities" ||
                      indicatorName === "Equity Before Minorities" ||
                      indicatorName === "Change in Working Capital" ? (
                      <td className="financialsTotalTD">
                        {value[indicatorName] ? value[indicatorName] : "-"}
                      </td>
                    ) : (
                      <td>
                        {value[indicatorName] ? value[indicatorName] : "-"}
                      </td>
                    )}
                  </>
                ))}
              </>
            </tr>
          ))}
        </table>
      </div>
    );
  }
}
