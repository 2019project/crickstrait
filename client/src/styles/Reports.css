@import url(https://fonts.googleapis.com/css?family=Karla);

* {
  font-family: Karla;
  scroll-behavior: smooth;
}

.financialTableContainer {
  padding: 0px 0px 0px 0px;
  width: 100%;
  overflow-x: scroll;
  box-shadow: 0px 1px 5px #2c3e50;
  margin-bottom: 20px;
  position: relative;
}

.financialTableDataTitleNames {
  background-color: #2c3e50 !important;
  color: #2c3e50;
}
.financialTableData {
  border-collapse: collapse;
}

.financialTableDataTitle {
  background-color: #2c3e50 !important;
  color: white;
}
.financialTableData td {
  text-align: center;
  font-size: 10px;
  padding: 8px 0px;
  min-width: 100px;
}
.financialTableData tr td:first-of-type {
  font-size: 12px;
  /* font-weight: bold; */
  min-width: 200px;
  position: sticky;
  left: 0px;
  background-color: white;
  
}
.financialsTotalTD {
  font-size: 12px !important;
  font-weight: bold;
}
 
@media only screen and (min-width: 426px) {
  .financialTableContainer {
    padding: 0px 0px 0px 0px;
    width: 100%;
    overflow-x: scroll;
    box-shadow: 0px 1px 10px #2c3e50;
    margin-bottom: 50px;
    position: relative;
  }

  .financialTableData td {
    text-align: center;
    font-size: 14px;
    padding: 5px 15px;
   }
  /* .financialTableContainer td:nth-child(even) {
    background-color: #f2f2f2;
  } */
  .financialTableData tr td:first-of-type {
    font-size: 15px;
    /* font-weight: bold; */
    min-width: 300px;
  }
  .financialsTotalTD {
    font-size: 16px !important;
    font-weight: bold;
  }
}
