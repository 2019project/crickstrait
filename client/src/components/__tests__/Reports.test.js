import React from "react";
import Reports from "../Reports";
import { shallow, mount } from "enzyme";

const reports = jest.fn();
const headers = [
    "date","test2","test3","Total Assets","Total Liabilities"
  ];
const reportdata = [{
    date:"123",
    test2:"123",
    test3:null,
    'Total Assets':"123",
    "Total Liabilities":null,

  }];

const wrapper = shallow(<Reports reports={reports} headers={headers} reportdata={reportdata} />);

describe("Testing Navbar Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
