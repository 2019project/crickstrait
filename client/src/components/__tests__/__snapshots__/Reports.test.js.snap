// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component should mount the component 1`] = `
<div
  className="financialTableContainer"
>
  <table
    className="financialTableData2"
  >
    <tr>
      <td
        className="financialTableDataTitle"
      >
        Indicator Name
      </td>
      <td
        className="financialTableDataTitle"
      >
         
        Jan 23
      </td>
    </tr>
    <tr>
      <td>
        test2
      </td>
      <td>
        123
      </td>
    </tr>
    <tr>
      <td>
        test3
      </td>
      <td>
        -
      </td>
    </tr>
    <tr>
      <td
        className="financialsTotalTD"
      >
        Total Assets
      </td>
      <td
        className="financialsTotalTD"
      >
        123
      </td>
    </tr>
    <tr>
      <td
        className="financialsTotalTD"
      >
        Total Liabilities
      </td>
      <td
        className="financialsTotalTD"
      >
        -
      </td>
    </tr>
  </table>
</div>
`;
